### ISSUE TITLE
Question: Got all nan loss when training softmatch

### ISSUE THREAD

**Issue by lhq12** [microsoft/Semi-supervised-learning#88]
Hello:
I have a problem when training SoftMatch. I run the code using the given config file: `python train.py --c Semi-supervised-learning/config/classic_cv/softmatch/cifar10_250_0.yaml --use_aim`. At its early stage, I found the sup_loss getting larger and unsup_loss geting smaller, and then all loss became `nan` after about 4000 iters.

Does anybody meet the same problem? I dont know if this is due to the SoftMatch algorithm or not.

**Comment by Hhhhhhao** [comment]
Have you changed the hyper-parameters or used amp? Can you post your training log here? 

I just ran the same config and setting on a single GPU, everything looks fine.

![image](https://user-images.githubusercontent.com/39571115/226077459-6d4101ca-4544-4454-9488-8f390f1436c2.png)

**Comment by lhq12** [comment]
Thanks for the timely reply. Here is the log file.
[log.txt](https://github.com/microsoft/Semi-supervised-learning/files/11007284/log.txt)

**Comment by Hhhhhhao** [comment]
Ok I found the reason. uratio needs to be set to 7 but not reflect on the config of freematch and softmatch, I will make a update on both configs.

**Comment by Hhhhhhao** [comment]
Fixed in 9288db0d34ea027a65596d6b7010c952b68b5a6b. Can you check again about the performance?

**Comment by lhq12** [comment]
Thanks! I will try it again!

**Comment by lhq12** [comment]
[log.txt](https://github.com/microsoft/Semi-supervised-learning/files/11009066/log.txt)
Sorry, the new config file does not work, either.

**Comment by Hhhhhhao** [comment]
I also noticed you are setting distributed and multiprocessing_distributed to False. Don't know if that is the reason causing the nan problem.

This is the exact code we used in the paper (except for configs), so If you are using the originally updated config you should be fine.

This is the log I just ran without changing anything in config after the fix.
![image](https://user-images.githubusercontent.com/39571115/226126006-8d46fefa-3098-4b6c-8e5c-e7911d0bee98.png)

**Comment by lhq12** [comment]
Thank you for your kind reply. I recloned the repo and it is fixed. Maybe I have changed some code by mistake.
And there is a puzzl left: is it necessary to train the model for 2^20 iters? It really takes time. It seems that the batch_size is in fact batch_size * number_of_GPU, will this affect the training result?

**Comment by Hhhhhhao** [comment]
For fair comparison we run all algorithms for 2^20 iterations. The batch size for all algorithms is also the same. 

If you use the algorithms for your own applications, it seems 25k or 50k total iterations is also fine.
